### PR TITLE
Add support for overzoom rendering of vector tiles

### PR DESCRIPTION
**Pull request by JesseCrocker**
This adds support for rendering when the max zoom for a style is greater than the max zoom for the vector tile source. Currently this is only supported for tilejson based sources.

### CODE REVIEW

File: src/back/VectorBasedTile.js [hunk 1]
```diff
@@ -13,7 +13,26 @@ util.inherits(VectorBasedTile, Tile);
...
-    var vtile = new mapnik.VectorTile(this.z, this.x, this.y),
+
+    //Support for overzooming
+    var maxzoom = 99,
+        params = {
+            z: this.z,
+            x: this.x,
+            y: this.y
+        };
+    for (var i = 0; i < project.mml.source.length; i++) {
```
Comment (yohanboniface) [inline src/back/VectorBasedTile.js]: Can't we do that only once on project load instead?

**Comment by JesseCrocker** [comment]
@yohanboniface I've changed it to only calculate the source max zoom once, and enabled overzoom rendering for local sources.

File: src/plugins/datasource-loader/index.js [hunk 2]
```diff
@@ -14,8 +14,17 @@ var DataSourceLoader = function (config) {
 DataSourceLoader.prototype.patchMML = function (e) {
     if (!e.project.mml) return e.continue();
     var processed = 0, self = this,
+        sourceMaxzoom = 100,
         sources = e.project.mml.source,
-        commit = function () {if (++processed === sources.length) e.continue();},
+        commit = function () {
+            if (++processed === sources.length){
+                for (var i = 0; i < sources.length; i++) {
```
Comment (yohanboniface) [inline src/plugins/datasource-loader/index.js]: I would move this bit of code in a separate method, giving it an understandable name, and then call it here.

**Comment by yohanboniface** [comment]
Let's proceed :)

**Comment by JesseCrocker** [comment]
@yohanboniface I was still planning on making that change, just been distracted on other things for the last few weeks. Let me know if you still want it changed.

**Pull request merged**